### ISSUE TITLE
Account for arrow head in arrow length

### ISSUE THREAD

**Issue by BambOoxX** [MakieOrg/Makie.jl#2472]
This issue is follows the discussion in this discourse [thread](https://discourse.julialang.org/t/makie-3d-arrows-scale-in-a-strange-fashion/89371/5).

### Problem
Currently, 3D arrows (2D was not tested) are plotted without accounting for their head. Which means that the total length of an arrow corresponds to the length of the tail + the length of the head. In a more practical fashion, that means that 
```
arrows!(ax,Point3f[(0,0,0)],Point3f[(1,0,0)],arrowsize = 0.5)
```
will result in an overall shorter arrow than
```
arrows!(ax,Point3f[(0,0,0)],Point3f[(1,0,0)],arrowsize = 1.0)
```
It could be nice to have an option (so as to not be a breaking change) to have the total length of the arrow account for the head length. 

### Example
```julia
using GLMakie

fig = Figure()
ax = LScene(fig[1,1])
# Has to be added otherwise first arrows with single arrow fails
scatter!(ax,Point3f[(2,0,0),(0,2,0),(0,0,2)])
# Effect of arrowsize parameter
arrows!(ax,Point3f[(0,0,0)],Point3f[(1,0,0)],color = RGBAf(255,0,0,1),arrowsize = 0.5)
arrows!(ax,Point3f[(0,0,0)],Point3f[(1,0,0)],color = RGBAf(0,255,0,0.5),arrowsize = 1)
# Just for fun effect of align parameter (surprising ?)
poss = (:head, :lineend, :tailend, :headstart, :center,:origin)
for (z,pos) in enumerate(poss)
arrows!(ax,Point3f[(0,0,z)],Point3f[(1,0,0)],color = RGBAf(0,0,255,0.5),align=pos,arrowsize = 1)
end
display(GLMakie.Screen(),fig)
```
![image](https://user-images.githubusercontent.com/42067365/205689637-82cd378b-b882-4f60-973a-0570357fb373.png)

While I was at it, a bit of fiddling with position alignment showed a surprising behavior.

**Comment by ffreyer** [comment]
Just a thought I wanted to put down somewhere: We can also create 2d arrows just with lines/linesegments by doing

```julia
linesegments([0, 1, 1, 2], [1,1,1,1], linewidth = [5,5,100,0])
```

With this though, the arrow length is given in data coordinates while the width is in pixel coordinates. 

We could also do a bunch of hacky stuff with linecaps and use them for arrows. This would involve:
- allowing start and end linecaps being adjusted independently
- linecap width being independent of linewidth
- linecap length being adjustable and allowing for negative lengths to cut into the line (which I currently have as part of pr)
- adding a triangle cap (which I had but removed)

With this approach we could get away with one line segment per arrow and all scales would be in pixel space. We'd also get cleaner tail-to-head boundaries. For CairoMakie we can use the first approach to get similar behavior.

---

On a less technical note - How should we actually handle sizes with the length constraint? Relative sizes don't seem desirable to me since that would mean shorter arrows get squished and longer ones get stretched. And absolute size for the head would mean that the tail vanishes at some point, and beyond that we'd again have the wrong length. I guess a mixture would be most desirable, where arrow heads are constant size until the line vanishes and then shrink?

**Comment by BambOoxX** [comment]
@ffreyer 
> On a less technical note - How should we actually handle sizes with the length constraint? Relative sizes don't seem desirable to me since that would mean shorter arrows get squished and longer ones get stretched. And absolute size for the head would mean that the tail vanishes at some point, and beyond that we'd again have the wrong length. I guess a mixture would be most desirable, where arrow heads are constant size until the line vanishes and then shrink?

I agree with you. At some point there is no good solution to this kind of problem. In any case, I think it's better to let the user have the choice, but not make too fancy stuff since this kind of _clever_ display may have an effect on data perception
